python/cudf/cudf/tests/test_datetime.py

```diff
@@ -362,7 +362,7 @@ def test_typecast_from_datetime_to_datetime(data, from_dtype, to_dtype):
 @pytest.mark.parametrize("data", [numerical_data()])
 @pytest.mark.parametrize("nulls", ["some", "all"])
 def test_to_from_pandas_nulls(data, nulls):
-    pd_data = pd.Series(data.copy())
+    pd_data = pd.Series(data.copy().astype('datetime64[ns]'))
```

Comment:
Need else the setting of half of the values to `NaT`s will implicitly convert the series to `Int` with `None`. With this change we actually get the desired datetime type series.